src/scripts/update-cache.ts

```diff
@@ -0,0 +1,553 @@
+import {
+  IGovernanceCore_ABI,
+  IGovernanceDataHelper_ABI,
```

Comment:
as written in the other review using data helpers, makes the script non agnostic and complicated/unfeasible to use in any other project as these addresses cannot be inferred from the governance contract.

Also the advantage of using them is absolutely minimal:
- for fetching the proposals with helpers it seems to be worse to me, the logic is getting more complicated with the end result of fetching more data then you need in a lot of cases: https://github.com/bgd-labs/aave-governance-ui-helpers/pull/39/files#diff-2cc01b91a772d41ecd9a44c959eb7250febe28863a0a1837fe833d0a5ab51956R433-R462
- for fetching payloads again there is close to zero benefit
- for fetching voting machine stuff, i didn't check, but assume it's similar to the above


I'd still highly recommend to not use the helpers on caching as it simply makes the caching non reusable for other projects and results in close to zero if not zero benefit.

---

as this file was renamed/ moved and the contents rearranged it's quite hard to review the overall changes.
One weird thing that i noticed was re-introduced was the support for multiple payloadsControllers per network, making the config & caching logic unnecessary complicated.

Perhaps @sendra or @kyzia551 can confirm again, but to me they confirmed in the past that there will only ever be one payloadsController per network. The only thing that could ever happen is that a payloadsController is replaced - meaning one get's removed in favor of another one, so even in this case no multi-payloadscontroller support is needed.

Reply:
I understand what you’re talking about, but the whole point is that I need data from the voting machine and payloads inside the file in order to formulate approximately the same data format as in the interface

if I use caching like you (your version), then I need to greatly complicate the data parser logic for the UI, which I don’t really like...

I don’t understand what’s stopping you from using this cache in other places?
You can just take `.json` data from this repository with a request (fetch), and work with this data, or what do you mean?

Reply:
`for fetching payloads again there is close to zero benefit` - when payload not in proposal, than make sense, if we don't need this, than zero benefit yes, but I think we need this

`for fetching voting machine stuff, i didn't check, but assume it's similar to the above` - maybe it can be put into a parser... yes, will take a look

Reply:
> I don’t understand what’s stopping you from using this cache in other places?
You can just take .json data from this repository with a request (fetch), and work with this data, or what do you mean?

There's basically two reasons:
1. Our other packages are agnostic and can in theory be used with any instance of governance ui v3, not only our deployments. Therefore using via this package is not feasible(as the cache will simply not exist).
2. Even when using for the same governance instances then the one we cache here, relying on cache from here doesn't make sense for us as:
  a) we need the cache to be as fresh as possible (not potentially hours old)
  b) even for performance reasons makes sense to just have a local copy of the fresh cache vs fetching from a different repo with no assurance on freshness
  
---

Your second message i don't understand.

From looking at these helper contracts:
- **govHelpers**: allows fetching multiple proposals at once (which is arguable increasing complexity on the fetching on caching) - in addition it fetches the voting chainId for each proposal which arguably is a bit useless and wasteful as it's immutable. Probably would be better to just cache the portals and lookup.
- **payloadsControllerHelpers**: does quite literally nothing but batching which for caching is a bit useless. same can be archived with multicall if one wanted, but in reality it's fast enough to just do sequentially
- **votingMachineHelpers**: this one is a bit more fightable as it does a few more things, but again most of these things could be fetched once and cached forever as they are immutable(warehouse & strategy). Some others only need to be fetched once per proposal (getProposalVoteConfiguration), some others only until the first state change occurs (hasRequiredRoots), afterwards you only need `getProposalState`*

* in this 3th case I would even agree, that using the helper has benefits. Still they don't outweigh the downside of making it impossible to use in some circumstance & more complicated to use in some others - especially when the workaround of not using them is sub 100 lines of code and might even slightly improve performance / decrease compute units.